src/ModuleA.lua

```diff
@@ -363,14 +363,14 @@ function M.build_availA(self)
    local function build_availA_helper(mpath, sn, v, A)
       local icnt = #A
       if (v.file ) then
-         if (show_hidden or mrc:isVisible(sn)) then
+         if (show_hidden or mrc:isVisible({fullName=sn,sn=sn,fn=v.file})) then
```

Comment:
You have it correct.  In this case the fullName and sn are the same.    This is the case where there is no version.  You can get this where MODULEPATH=/apt/mfiles:... 
and /apt/mfiles/TACC is a modulefile.  In this case fullName=TACC sn=TACC.